pwiz_tools/Skyline/SkylineNightly/LogFileMonitor.cs

```diff
@@ -88,26 +92,39 @@ public void Stop()
         private int HangThreshold => _runMode != Nightly.RunMode.perf && _runMode != Nightly.RunMode.release_perf && _runMode != Nightly.RunMode.integration_perf ? 60 : 90;
         private string RunModeName => Enum.GetName(typeof(Nightly.RunMode), _runMode);
 
-        public bool ExtendNightlyEndTime
+        public bool IsHang
         {
             get
             {
-                try
+                lock (_lock)
                 {
-                    return File.GetLastWriteTime(_testerLog).Equals(_lastReportedHang);
+                    return _isHang;
                 }
-                catch
+            }
+        }
+
+        public bool IsDebugger
+        {
+            get
+            {
+                if (_debuggerAttached)
+                    return true;
+
+                CheckLog(null, null);
+                lock (_lock)
                 {
-                    return false;
+                    return _debuggerAttached;
                 }
             }
         }
 
-        private void IntervalElapsed(object source, ElapsedEventArgs e)
+        private void CheckLog(object source, ElapsedEventArgs e)
         {
             if (!Monitor.TryEnter(_lock))
                 return;
 
+            var signalTime = e?.SignalTime ?? DateTime.Now;
```

Comment:
The lock is probably not necessary but I think the timer event runs on a separate thread. So in the case where it doesn't finish running before the next interval, I don't want it running twice concurrently.

Reply:
> The lock is probably not necessary but I think the timer event runs on a separate thread. So in the case where it doesn't finish running before the next interval, I don't want it running twice concurrently.

Oh, right. I was confused.
Timer controls fire on the same thread that they were created on, but regular timers don't.

It's still a little weird that you're doing "Monitor.TryEnter" instead of just a regular "lock()". Usually you don't care enough about waiting until the other lock holder is done for it to be worth splitting it into a "Monitor.TryEnter" and "Monitor.Exit".